graphene_django/fields.py

```diff
@@ -103,39 +102,44 @@ def resolve_connection(cls, connection, default_manager, args, iterable):
 
     @classmethod
     def connection_resolver(
-        cls,
-        resolver,
-        connection,
-        default_manager,
-        max_limit,
-        enforce_first_or_last,
-        root,
-        info,
-        **args
-    ):
-        first = args.get("first")
-        last = args.get("last")
-
-        if enforce_first_or_last:
-            assert first or last, (
-                "You must provide a `first` or `last` value to properly paginate the `{}` connection."
-            ).format(info.field_name)
-
-        if max_limit:
-            if first:
-                assert first <= max_limit, (
-                    "Requesting {} records on the `{}` connection exceeds the `first` limit of {} records."
-                ).format(first, info.field_name, max_limit)
-                args["first"] = min(first, max_limit)
-
-            if last:
-                assert last <= max_limit, (
-                    "Requesting {} records on the `{}` connection exceeds the `last` limit of {} records."
-                ).format(last, info.field_name, max_limit)
-                args["last"] = min(last, max_limit)
-
-        iterable = resolver(root, info, **args)
-        on_resolve = partial(cls.resolve_connection, connection, default_manager, args)
+            cls,
+            resolver,
+            connection,
+            default_manager,
+            max_limit,
+            enforce_first_or_last,
+            root,
+            info,
+            **kwargs):
+        # pylint: disable=R0913,W0221
+
+        first = kwargs.get("first")
+        last = kwargs.get("last")
+        if not (first is None or first > 0):
+            raise ValueError(
+                "`first` argument must be positive, got `{first}`".format(**locals()))
```

Comment:
The call to locals and constructing the function call to format is fairly expensive here I think. Could we instead pass the kwarg we need? (Same in the other instances in this function)